blockchain/difficulty.go

```diff
@@ -540,6 +540,58 @@ func (b *BlockChain) sumPurchasedTickets(startNode *blockNode, numToSum int64) i
 	return numPurchased
 }
 
+// immatureTickets returns the total number of immature tickets at the
+// current tip and a map of immature ticket counts keyed by the block height at
+// which they were bought at.
+func (b *BlockChain) immatureTickets() (int64, map[int64]int64) {
+	tip := b.bestChain.Tip()
+	var numImmature int64
+	immatureSet := make(map[int64]int64, b.chainParams.TicketMaturity)
+	for node, numTraversed := tip, int64(0); node != nil &&
+		numTraversed <= int64(b.chainParams.TicketMaturity); numTraversed++ {
```

Comment:
Shouldn't this be `<` to avoid double-counting mature tickets?

To illustrate where I'm coming from, let's assume `TicketMaturity = 1` and `tip` has a height of 10. A ticket included in block 9 would mature in block 10 and therefore would not be immature. The nuance here is that despite block 9's ticket purchases being mature in block 10 the following constraints are in play that prevent it from voting until block 11:

* a utxo must precede its redeemer
* votes must precede tickets in a block
* the lock time relative to the block the tx was mined in must elapse before it is spendable (TicketMaturity)

Regardless if there is an off-by-one scenario here or not, there is room for a comment explaining what's going on and why for future devs since this part of the code is really tricky due to the non-obvious relation between transaction ordering in a block and maturity requirements defined elsewhere.

Reply:
Tickets mature at `TicketMaturity` but aren't immediately spendable, they become spendable at `TicketMaturity+1`, referring to the example provided the ticket included in block 9 would mature in block 10 and be available to vote in block 11. 

Thanks for pointing this out, will update with comments.